### PR TITLE
Seet main merge

### PR DESCRIPTION
**Pull request by egull**
This is not yet to be merged, WIP.

### CODE REVIEW

**Comment by egull** [comment]
Now ready to review. This should bring the full symmetry functionality from main to SEET. Propose to immediately after merge SEET back to main (and discontinue using SEET branch).

Points to review carefully:
1. updated test files (conforming to the new symmetry framework)
2. updated dependency (green-impurity) with hdf5 also updated for new symmetry framework.

Claude was unhappy about boundary values in the self-energy that seemed incorrect but untested. New test seems to capture them properly.

**Comment by egull** [comment]
Adding @yangyu-code and @sdirnboeck  as optional reviewers, because some of this work was due to them.

**Comment by egull** [comment]
[also note that tests will start passing once https://github.com/Green-Phys/green-impurity/pull/11 has been merged]

**Review by agnieszkajazdzewska — APPROVED** [review]

File: src/green/mbpt/mbpt_run.h [hunk 1]
```diff
@@ -237,16 +237,16 @@ namespace green::mbpt {
...
-    std::string path = p["input_file"];
+  inline void check_input(const params::params& p) {
+    std::string   path = p["input_file"];
     h5pp::archive ar(path, "r");
-    if(ar.has_attribute("__green_version__")) {
+    if (ar.has_attribute("__green_version__")) {
       std::string int_version = ar.get_attribute<std::string>("__green_version__");
-      if (!CheckVersion(int_version)) {
-        throw mbpt_outdated_input("Input file at '" + path +"' is outdated, please run migration script python/migrate.py");
+      if (int_version.rfind(INPUT_VERSION, 0) != 0) {
```
Comment (gauravharsha) [inline src/green/mbpt/mbpt_run.h]: `int_version.rfind` is not suitable for testing version compatibilities anymore. The correct way is to use the main branch:
```
if (!CheckVersion(int_version)) {
...
}
```

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -1,7 +1,17 @@
 project(mbpt-lib CXX)
 
+# Weak coupling many-body framework
 add_library(mbpt common_utils.cpp dyson.cpp hf_solver.cpp gw_solver.cpp gf2_solver.cpp gf2_solver_t_ewald_correction.cpp mbpt_q0_utils_t.cpp
         hf_cpu_kernels.cpp
         gw_cpu_kernel.cpp)
 target_include_directories(mbpt PUBLIC .)
-target_link_libraries(mbpt PUBLIC GREEN::PARAMS GREEN::H5PP GREEN::UTILS GREEN::NDARRAY GREEN::GRIDS GREEN::SYMMETRY GREEN::SC)
\ No newline at end of file
+target_link_libraries(mbpt PUBLIC GREEN::PARAMS GREEN::H5PP GREEN::UTILS GREEN::NDARRAY GREEN::GRIDS GREEN::SYMMETRY GREEN::SC GREEN::IMPURITY)
```
Comment (gauravharsha) [inline src/CMakeLists.txt:8]: Not sure if we need `GREEN::IMPURITY`to be linked with `mbpt.exe`.

File: src/gw_cpu_kernel.cpp [hunk 3]
```diff
@@ -29,15 +29,21 @@
 namespace green::mbpt::kernels {
 
   void gw_cpu_kernel::solve(G_type& g, St_type& sigma_tau) {
-    _coul_int1 = new df_integral_t(_path, _nao, _NQ, _bz_utils);
-    utils::shared_object<ztensor<4>> P0_tilde_s(_nts, 1, _NQ, _NQ);
-    utils::shared_object<ztensor<4>> Pw_tilde_s(_nw_b, 1, _NQ, _NQ);
+    auto cntx = g.cntx();
+#ifndef NDEBUG
```
Comment (gauravharsha) [inline src/gw_cpu_kernel.cpp]: Any specific reason for the `#ifndef NDEBUG` and assertions?

File: CMakeLists.txt [hunk 4]
```diff
@@ -8,9 +8,11 @@ if (${CMAKE_CURRENT_SOURCE_DIR} STREQUAL ${CMAKE_CURRENT_BINARY_DIR})
 endif ()
 
 ################ GREEN RELEASE VERSION #####################
-file(READ ${CMAKE_SOURCE_DIR}/GREEN_RELEASE.txt GREEN_RELEASE_DEFAULT)
-string(STRIP ${GREEN_RELEASE_DEFAULT} GREEN_RELEASE_DEFAULT)
-set (GREEN_RELEASE ${GREEN_RELEASE_DEFAULT} CACHE STRING "Release version for Green MBPT")
+set(GREEN_RELEASE origin/main CACHE STRING "Release version for Green dependencies")
```
Comment (gauravharsha) [inline CMakeLists.txt]: This can now be restored to GREEN_RELEASE.txt.
The idea is that GREEN_RELEASE.txt defaults to `origin/main`. And when a new version is released, it automatically triggers the creation of new tags on the dependent modules.

File: sim.h5
Comment (gauravharsha) [inline sim.h5:1]: I think this is one of the files wrongly committed to the repo (likely by m?). Should be removed.

**Review by gauravharsha — APPROVED** [review]
Minor changes, but otherwise looks good. Tests are also passing.

File: src/CMakeLists.txt (hunk 2 above)
Comment (egull) [reply src/CMakeLists.txt]: Actually it doesn't work if we remove this dependence, due to green/embedding/seet_solver.h using impurity::impurity_solver. @gauravharsha please check if that makes sense to you.
